contracts/IdleCDOAutoLossVariant.sol

```diff
@@ -0,0 +1,190 @@
+  /// @return _virtualPrice tranche price considering all interest
+  /// @return _totalTrancheGain (int256) tranche gain/loss since last update
+  function _virtualPriceAuxVariant(
+    address _tranche,
+    uint256 _nav,
+    uint256 _lastNAV,
+    uint256 _lastTrancheNAV,
+    uint256 _trancheAPRSplitRatio
+  ) internal view returns (uint256 _virtualPrice, int256 _totalTrancheGain) {
+    // Check if there are tranche holders
+    uint256 trancheSupply = IdleCDOTranche(_tranche).totalSupply();
+    if (_lastNAV == 0 || trancheSupply == 0) {
+      return (oneToken, 0);
+    }
+
+    // In order to correctly split the interest generated between AA and BB tranche holders
+    // (according to the trancheAPRSplitRatio) we need to know how much interest/loss we gained
+    // since the last price update (during a depositXX/withdrawXX/harvest)
+    // To do that we need to get the current value of the assets in this contract
+    // and the last saved one (always during a depositXX/withdrawXX/harvest)
+    // Calculate the total gain/loss
+    int256 totalGain = int256(_nav) - int256(_lastNAV);
+    // If there is no gain/loss return the current price
+    if (totalGain == 0) {
+      return (_tranchePrice(_tranche), 0);
+    }
+
+    // Remove performance fee for gains
+    if (totalGain > 0) {
+      totalGain -= totalGain * int256(fee) / int256(FULL_ALLOC);
+    }
+
+    address _AATranche = AATranche;
+    address _BBTranche = BBTranche;
+    bool _isAATranche = _tranche == _AATranche;
+    // Get the supply of the other tranche and
+    // if it's 0 then give all gain to the current `_tranche` holders
+    if (IdleCDOTranche(_isAATranche ? _BBTranche : _AATranche).totalSupply() == 0) {
+      _totalTrancheGain = totalGain;
+    } else {
+      if (totalGain > 0) {
+        // Split the net gain, with precision loss favoring the AA tranche.
+        int256 totalBBGain = totalGain * int256(FULL_ALLOC - _trancheAPRSplitRatio) / int256(FULL_ALLOC);
+        // The new NAV for the tranche is old NAV + total gain for the tranche
+        _totalTrancheGain = _isAATranche ? (totalGain - totalBBGain) : totalBBGain;
+      } else { // totalGain is negative here
+        // Redirect the whole loss (which should be < maxDecreaseDefault) to junior holders
+        int256 _juniorTVL = int256(_isAATranche ? _lastNAV - _lastTrancheNAV : _lastTrancheNAV);
+        int256 _newJuniorTVL = _juniorTVL + totalGain; 
+        // if junior holders have enough TVL to cover
+        if (_newJuniorTVL > 0) {
+          _totalTrancheGain = _isAATranche ? int256(0) : totalGain;
+        } else {
+          // otherwise all loss minus junior tvl to senior
+          if (!_isAATranche) {
+            // juniors have no more claim (we set price to *1 wei* to avoid 0), gain is set to -juniorTVL
+            return (1, -_juniorTVL);
+          }
+          // seniors get the loss - oldjuniorTVL (- *1 wei* for each tranche junior tranche token)
```

Comment:
question: why do we set 1 wei for price  here.

Reply:
I wanted to avoid having a 0 price as it may cause issues eg here ```_minted = _amount * ONE_TRANCHE_TOKEN / _tranchePrice(_tranche);``` when we mint new shares, but deposits are actually disabled and it makes more sense to have it equal to 0. I'll update the code